spec/controllers/answer_tags_controller_spec.rb

```diff
@@ -10,6 +10,16 @@
   let!(:tag_prompt_deployment2) { create(:tag_prompt_deployment, id: 2, assignment_id: 2, questionnaire_id: 1) }
   let!(:answer_tag) { create(:answer_tag, id: 1, tag_prompt_deployment_id: 1, user_id: student.id) }
 
+
+  #factory objects required for "create_edit" test cases - since creating answer tags and updating answer tags requires pre mapping of answer and tag deployment key constraints
+  let(:questionnaire1) { create(:questionnaire, id: 2) }
+  let(:question1) { create(:question, questionnaire: questionnaire, weight: 2, id: 2, type: "Criterion") }
+  let(:response_map) { create(:review_response_map, id: 2, reviewed_object_id: 2) }
+  let!(:response_record) { create(:response, id: 2, response_map: response_map) }
+  let!(:answer) { create(:answer, question: question1, comments: "test comment", response_id: response_record.id) }
+  let(:tag_prompt) { TagPrompt.create id: 3, prompt: "??", desc: "desc", control_type: "slider" }
+  let(:tag_deploy) { TagPromptDeployment.create id: 3, tag_prompt: tag_prompt, question_type: "Criterion" }
```

Comment:
Similar to the previous line.
```suggestion
  let(:tag_deploy) { create(:tag_prompt_deployment, id: 3, tag_prompt: tag_prompt, question_type: "Criterion") }
```